src/analysis/calculus/conformal.lean

```diff
@@ -0,0 +1,156 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import analysis.normed_space.conformal_linear_map
+
+/-!
+# Conformal Maps
+
+A continuous linear map between real normed spaces `X` and `Y` is `conformal_at` some point `x`
+if it is real differentiable at that point and its differential `is_conformal_linear_map`.
+
+## Main results
+
+* `conformal_at`: the main definition
+* The conformality of the composition of two conformal maps, the identity map
+  and multiplications by nonzero constants
+* `conformal_at_iff`: an equivalent definition of the conformality
+* `conformal_at.preserves_angle`: if a map is conformal at `x`, then its differential
+                                  preserves all angles at `x`
+* `conformal_factor_at`: the conformal factor of a conformal map at some point
+
+## Tags
+
+conformal
+
+## Warning
+
+The definition of conformality in this file does NOT require the maps to be orientation-preserving.
+Maps such as the complex conjugate is considered as a conformal map.
+-/
+
+noncomputable theory
+
+variables {E F : Type*} [inner_product_space ℝ E] [inner_product_space ℝ F]
+  {X Y Z : Type*} [normed_group X] [normed_group Y] [normed_group Z]
+  [normed_space ℝ X] [normed_space ℝ Y] [normed_space ℝ Z]
+
+section loc_conformality
+
+open linear_isometry continuous_linear_map
+open_locale real_inner_product_space
+
+/-- A map `f` is said to be conformal if it has a conformal differential `f'`. -/
+def conformal_at (f : X → Y) (x : X) :=
+∃ (f' : X →L[ℝ] Y), has_fderiv_at f f' x ∧ is_conformal_map f'
+
+lemma conformal_at_id (x : X) : conformal_at id x :=
+⟨id ℝ X, has_fderiv_at_id _, 1, one_ne_zero, id, by ext; simp⟩
+
+lemma conformal_at_const_smul {c : ℝ} (h : c ≠ 0) (x : X) :
+  conformal_at (λ (x': X), c • x') x :=
+⟨c • continuous_linear_map.id ℝ X,
+  has_fderiv_at.const_smul (has_fderiv_at_id x) c, is_conformal_map_const_smul h⟩
+
+lemma conformal_at_of_conformal_fderiv {f : X → Y} {x : X}
+  (h : differentiable_at ℝ f x) (H : is_conformal_map (fderiv ℝ f x)) : conformal_at f x :=
```

Comment:
Don't require `differentiable_at` in this lemma ... I know it's a pain to prove, but it's much more useful without!